pages/__app/forgotPassword.tsx

```diff
@@ -0,0 +1,144 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { LucideHeartCrack } from 'lucide-react'
+import type { NextPage } from 'next'
+import Image from 'next/image'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { toast } from 'react-hot-toast'
+import * as z from 'zod'
+
+import { Button } from '@/components/ui/button'
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { useAuth } from '@/context/Authentication'
+import { useCurrentUser } from '@/hooks/query/user'
+import { onGraphqlErrorToast } from '@/lib/error'
+
+const formSchema = z.object({
+  email: z.string().email({ message: 'Invalid email' }),
+  password: z
+    .string()
+    .min(8, { message: 'Password must be at least 8 characters long' }),
+})
+
+const ForgotPasswordPage: NextPage = () => {
+  const [loading, setLoading] = useState(false)
+
+  const router = useRouter()
+
+  const { signInWithEmailAndPassword } = useAuth()
+  const { user, isLoading: currentUserLoading } = useCurrentUser()
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      email: '',
+      password: '',
+    },
+  })
+
+  useEffect(() => {
+    if (user && user.id) {
+      router.replace('/dashboard')
+    }
+  }, [router, user])
+
+  const onSubmit = useCallback(
+    async (values: z.infer<typeof formSchema>) => {
+      setLoading(true)
+      try {
+        toast.loading('Please wait...', { id: 'signin-loading' })
+        if (signInWithEmailAndPassword)
+          await signInWithEmailAndPassword({
```

Comment:
will this API work to send a forgot password email?

Reply:
yess

Reply:
How forgot password email API can take password? when you do not have forgotten your password?

Reply:
Well, my bad I didn't see the code properly..
This is not going to work!

Reply:
@ruhani2 Thanks for your review, I saw the mistake.